Submission/sprint6

noxpardalis opened the pull request.

noxpardalis committed:
Remove extraneous timetracking directory

noxpardalis committed:
Merge pull request #17 from RUGSoftEng/doc/architecture

Doc/architecture

noxpardalis committed:
Remove extraneous time tracking directory

noxpardalis committed:
Merge branch 'docs' of github.com:RUGSoftEng/2018-Hestia-Web into docs

noxpardalis committed:
Merge branch 'frontend' into docs

noxpardalis committed:
Merge pull request #20 from RUGSoftEng/docs

Docs

Commit:
Merge pull request #21 from RUGSoftEng/frontend

Frontend merge

andrewlalis committed:
Added architecture diagram.

andrewlalis committed:
Added some images for slides, put them in architecture just in case we need them later.

andrewlalis committed:
Merged sprint2 data into the apache server version.

andrewlalis committed:
Re-organized files for apache version; no more static folder.

andrewlalis committed:
Created firebase scripts directory.

andrewlalis committed:
Removed firebase script from index.html

andrewlalis committed:
Moved some functions to eventHandler script, renamed functions.js to serverComms.js

andrewlalis committed:
Added domInteraction.js for all functions which modify the webpage.

andrewlalis committed:
Removed resources directory.

andrewlalis committed:
Got the requests working in PHP after much pain.

andrewlalis committed:
Floating points finally work

andrewlalis committed:
Posting works, refreshing is fine, fixed the issue with the selection reverting to the first element.

andrewlalis committed:
Removed obsolete getDevice() function.

andrewlalis committed:
Removed setup files, requirements.txt, and edited readme to tell the user to install apache, php, and curl, and configure a site. Perhaps I will make a script to install those for the user, but now it is not important.

andrewlalis committed:
Updated timesheet for myself!

andrewlalis committed:
Merge branch 'docs' of https://github.com/RUGSoftEng/2018-Hestia-Web into docs

andrewlalis committed:
Added server protocol image.

andrewlalis committed:
Removed unnecessary functions from auth.js

andrewlalis committed:
Added old python files back to apache version.

andrewlalis committed:
Added flask style directories, removed request.php

andrewlalis committed:
Updated launch script, added a template file and resource import file, and new index.html

andrewlalis committed:
Made working index and login pages with jinja2/flask.

andrewlalis committed:
Made devices list work properly on flask again.

andrewlalis committed:
Fixed current menu button icon.

andrewlalis committed:
Updated my timetracking

andrewlalis committed:
Added .gitignore to ignore pycache from flask.

andrewlalis committed:
Added gitignore.

andrewlalis committed:
Added functions to serverComms.js to delete and rename a device.

andrewlalis committed:
Improved firebase functions to retrieve and add servers.

andrewlalis committed:
Implemented basic firebase server functionality.

andrewlalis committed:
Renamed server to more appropriate serverAddress in serverComms.js

andrewlalis committed:
Added functions to get server collections and plugins.

andrewlalis committed:
Added getPluginInfo() to get plugin information from server.

andrewlalis committed:
Added plugin and collection selectors.

andrewlalis committed:
Renamed 'side light' header to Activators.

andrewlalis committed:
Add device now implemented.

andrewlalis committed:
Added alphabetically sorting activators by name.

noxpardalis committed:
Format code to be pep8 compliant

noxpardalis committed:
Merge pull request #23 from RUGSoftEng/flaskFromApacheVersion

Flask from apache version

noxpardalis committed:
Merge pull request #24 from RUGSoftEng/gitIgnore

Added .gitignore to ignore pycache folder.

andrewlalis committed:
Added timeouts to requests to hestia servers.

andrewlalis committed:
Amended timeout to 0.01s to allow slower connections.

noxpardalis committed:
Merge pull request #25 from RUGSoftEng/feature/addNewDevice

Add ability to add new device. Auto generated via Hestia Server output

noxpardalis committed:
Merge pull request #27 from RUGSoftEng/feature/serverRequestTimeouts

Feature/server request timeouts

noxpardalis committed:
Remove unneeded design images in site

andrewlalis committed:
Merge pull request #28 from RUGSoftEng/feature/removeFiles

Feature/remove files

noxpardalis committed:
Fix requirements.txt

noxpardalis committed:
Remove old html templates

andrewlalis committed:
Fixed broken server devices list.

andrewlalis committed:
Fixed zero and one floating point errors.

EliasAbdo committed:
Update S3004910_Elias_Abdo.md

EliasAbdo committed:
Update S3004910_Elias_Abdo.md

andrewlalis committed:
Merge pull request #29 from RUGSoftEng/feature/fixFlask

Fix requirements.txt

GarryBell committed:
addedn overview diagram from slides

GarryBell committed:
added file for legal stuff

Commit:
added files for legal matter

Commit:
some content and research done

noxpardalis committed:
Add functionality to add a server via the web gui

rtnijman committed:
Merge pull request #30 from RUGSoftEng/feature/addServersGUI

Add functionality to add a server via the web gui

noxpardalis committed:
Add functionality to delete a device from a server via the web GUI

rtnijman committed:
Merge pull request #31 from RUGSoftEng/deleteDevicesGUI

Add functionality to delete a device from a server via the web GUI

rtnijman committed:
Merge pull request #32 from RUGSoftEng/feature/fixFlask

Remove old html templates

poetinger committed:
Update timesheet for last week

Added two days of work
 Please enter the commit message for your changes. Lines starting

EliasAbdo committed:
Update S3004910_Elias_Abdo.md

EliasAbdo committed:
Update S3004910_Elias_Abdo.md

poetinger committed:
Added server deletion

Added simple button to remove server

noxpardalis committed:
Merge branch 'development' into feature/serversDisplay

Commit:
Merge pull request #33 from RUGSoftEng/feature/serversDisplay

Fixed broken server devices list.

GarryBell committed:
Merge branch 'development' of https://github.com/RUGSoftEng/2018-Hestia-Web into development

noxpardalis committed:
Add changes so devices display on slow connection

poetinger committed:
Add time and fixed dates

Added todays meeting, and fixed the dates of last commit.

SytseOegema committed:
Update S3173267_Sytse_Oegema.md

SytseOegema committed:
Update S3173267_Sytse_Oegema.md

GarryBell committed:
first commit i.o.t I can get it from my laptop

GarryBell committed:
added css to login button

GarryBell committed:
timetracking for 18/04 sprint

Commit:
Add rename

Adds a button that renames a device to Success.

Commit:
Attempted to fix bug

this.parentNode is undefined on line 331. But there IS NO this.parentNode in the code.

rtnijman committed:
Update S3222268_Rens_Nijman.md

Commit:
Fixed renaming device

Troy assisted with bug splat. Next it needs a input box.

Commit:
Add prompt for new device name

Added a nice prompt for the new name. Works.

Commit:
Fix server deletion

Fixed a null bug.

noxpardalis committed:
Merge pull request #34 from RUGSoftEng/feature/listServers

Feature/list servers

noxpardalis committed:
Merge pull request #35 from RUGSoftEng/feature/floatingFix

Fixed zero and one floating point errors.

noxpardalis committed:
Merge pull request #36 from RUGSoftEng/feature/logoutFix

Feature/logout fix

noxpardalis committed:
Ensure that the float fix is only performed on floats

Commit:
Merge pull request #38 from RUGSoftEng/feature/floatHotFix

Ensure that the float fix is only performed on floats

Commit:
Merge pull request #37 from RUGSoftEng/docs

Docs

Commit:
Last time update for sprint

Added evenings work

noxpardalis committed:
Update time tracking

noxpardalis committed:
Merge branch 'development' into timetracking

noxpardalis committed:
Merge pull request #39 from RUGSoftEng/timetracking

Timetracking

noxpardalis committed:
Remove unused concept site

noxpardalis committed:
Make python code pep8 compliant

Commit:
Merge pull request #40 from RUGSoftEng/feature/sprintCleaning

Feature/sprint cleaning

SytseOegema committed:
Update S3173267_Sytse_Oegema.md

andrewlalis committed:
Updated my timetracking, finally.

andrewlalis committed:
Added beginning of marketplace to architecture document.

noxpardalis committed:
Add first transition to RESTPlus for backend

noxpardalis committed:
Add readme for backend

noxpardalis committed:
Update '/request' endpoint description

noxpardalis committed:
Reflow requirements.txt

andrewlalis committed:
Added to marketplace section.

andrewlalis committed:
Updated timetracking.

poetinger committed:
Delete old timetacking

A ghost of the old timetracking system appeared after merging, this
removes it (it is in the up to date ones).

poetinger committed:
Added basic database

Basic database that can view contents of a database (must be running).

poetinger committed:
Change main to database

Just to reduce confusion, renamed main to database

noxpardalis committed:
Remove main.py

poetinger committed:
Add an update

I forget what I changed

poetinger committed:
Clean files

Cleaned up files a bit

poetinger committed:
Update timetracking

Added todays meeting and work to timesheet.

noxpardalis committed:
Switch to RESTPlus with integration with SQLAlchemy

noxpardalis committed:
Merge branch 'feature/flask_SQLAlchemy' of github.com:RUGSoftEng/2018-Hestia-Web into feature/flask_SQLAlchemy

noxpardalis committed:
Add port to model and database

poetinger committed:
Add documentation

Addes installation documentation for the flask side for both linux and windows.

noxpardalis committed:
Update README.md

noxpardalis committed:
Remove test application file

noxpardalis committed:
Update readme with description of setting up postgres via docker

noxpardalis committed:
Add pylint to project

noxpardalis committed:
Restructure packages

noxpardalis committed:
Update readme

noxpardalis committed:
Install and enable CORS

poetinger committed:
Add endpoints for user

Added user table. User table has userID and serversID. It is not yet complete and is NOT ready for use. I will fix afterlecture.

poetinger committed:
Add endpoints for user

Added user table. User table has userID and serversID. It is not yet complete and is NOT ready for use. I will fix afterlecture.

Commit:
done privacy policy, got most of T&C down

poetinger committed:
Merge branch 'feature/flask_SQLAlchemy' of https://github.com/RUGSoftEng/2018-Hestia-Web into feature/flask_SQLAlchemy

poetinger committed:
Fix client typos

The client was importing server, and server importing client. This is now fixed. Retrieving users is working. Posting is missing a required 'created_by'.

noxpardalis committed:
Remove created_by arguments for creating a user in the database

noxpardalis committed:
Fix case in database access

poetinger committed:
Add relationship work in progress

Adds relationships. Not complete.

poetinger committed:
Quick fix

Tables seem to need to exist before starting. Looking into now.

noxpardalis committed:
Add one-to-many relationship and switch to using strings as ids

poetinger committed:
Add more to fetch server

Getting closer to fully functioning. Still needs to get the actual data from the payload to fetch from the database.

noxpardalis committed:
Add dummy initdb command

noxpardalis committed:
Add newline

noxpardalis committed:
Revert addition of getServer route

noxpardalis committed:
Reintroduce original endpoints

poetinger committed:
Add time

Added this weeks current times

noxpardalis committed:
Add endpoints for user

EliasAbdo committed:
Update S3004910_Elias_Abdo.md

SytseOegema committed:
Update S3173267_Sytse_Oegema.md

SytseOegema committed:
Update S3173267_Sytse_Oegema.md

poetinger committed:
Add debug returns

noxpardalis committed:
Update with working deletion on users

noxpardalis committed:
Add server deletion

noxpardalis committed:
Readd ability to request information from remote controller

poetinger committed:
Clean code, added comments

First pass of pylint fixing.

noxpardalis committed:
Add pylint configuration

noxpardalis committed:
Update max-args in pylint from 5 to 6

poetinger committed:
Began updating the README to reflect the changes to backend

This will give detailed descriptions of the backend. WIP

poetinger committed:
Merge branch 'feature/flask_SQLAlchemy' of https://github.com/RUGSoftEng/2018-Hestia-Web into feature/flask_SQLAlchemy

noxpardalis committed:
Disable <no-self-use> and <too-few-public-methods>

noxpardalis committed:
Fix most pylint complaints

noxpardalis committed:
Merge pull request #42 from RUGSoftEng/feature/flask_SQLAlchemy

Feature/flask sql alchemy

poetinger committed:
Fix minor readme

Minor. Didn't even happend. You didn't see anything *waves hands*

rtnijman committed:
add files to build authentication

noxpardalis committed:
Update pipfile to prepare for integrating authentication

noxpardalis committed:
Add jose dependency

noxpardalis committed:
Delete unnecessary files

noxpardalis committed:
Add authentication boilerplate and require authentication for routes

noxpardalis committed:
Add ability to access endpoints only when passing correct authorization token

noxpardalis committed:
Fix authentication in routes and add method of authenticating in SWAGGER

noxpardalis committed:
Remove superfluous files

rtnijman committed:
add correct pip requirements

noxpardalis committed:
Remove env folder

SytseOegema committed:
Merge pull request #43 from RUGSoftEng/feature/auth0_login

Feature/auth0 login

Commit:
started on this spritnt's chacnges

rtnijman committed:
allow getting user_id and potentially more info

SytseOegema committed:
Updated the ubuntu installation guide lines.

SytseOegema committed:
Merge branch 'feature/flask_restplus' of github.com:RUGSoftEng/2018-Hestia-Web into feature/flask_restplus

rtnijman committed:
clean up unused user_ids

noxpardalis committed:
Remove unnecessary server_ids

noxpardalis committed:
Add broken transition to getting user_id for post to User from authentication header

Commit:
added work from laptop

Commit:
read before merge, still need to talk about Vue, otherwise reflects new design choices

SytseOegema committed:
Update S3173267_Sytse_Oegema.md

Update timetracking

rtnijman committed:
add functional but unfinished user post

NikDijkema committed:
Update S3230007_Nik_Dijkema.md

noxpardalis committed:
Add posting a server for a given user and getting servers for a particular user

noxpardalis committed:
Add filtration for individual servers based on current authenticated user

poetinger committed:
Update timesheet

Updated timesheet for sprint 4

noxpardalis committed:
Add automatic server id generation and remove expected server_id from endpoints

noxpardalis committed:
Remove getting users (is not currently necessary)

GarryBell committed:
my timetracking

GarryBell committed:
Merge branch 'timetracking' of https://github.com/RUGSoftEng/2018-Hestia-Web into timetracking

noxpardalis committed:
Merge pull request #44 from RUGSoftEng/feature/flask_restplus

Feature/flask restplus

rtnijman committed:
Update S3222268_Rens_Nijman.md

noxpardalis committed:
Merge branch 'docs' into doc/architecture

noxpardalis committed:
Merge pull request #45 from RUGSoftEng/doc/architecture

Doc/architecture

noxpardalis committed:
Merge pull request #46 from RUGSoftEng/timetracking

Timetracking

rtnijman committed:
Merge pull request #47 from RUGSoftEng/docs

Docs

rtnijman committed:
Update S3222268_Rens_Nijman.md

noxpardalis committed:
Update time tracking.

noxpardalis committed:
Merge branch 'timetracking' of github.com:RUGSoftEng/2018-Hestia-Web into timetracking

noxpardalis committed:
Merge pull request #49 from RUGSoftEng/timetracking

Update Timetracking

noxpardalis committed:
Merge pull request #50 from RUGSoftEng/development

Development

noxpardalis committed:
Perform fixes

noxpardalis committed:
Fix documentation

SytseOegema committed:
Merge pull request #51 from RUGSoftEng/feature/code_cleanup

Feature/code cleanup

EliasAbdo committed:
Merge pull request #52 from RUGSoftEng/development

Development

poetinger committed:
Add config for travis

poetinger committed:
Add cache of pip

poetinger committed:
Add dummy test

poetinger committed:
Add pipenv shell to install

poetinger committed:
Attempt to fix travis hang on shell

noxpardalis committed:
Merge branch 'feature/testing' of github.com:RUGSoftEng/2018-Hestia-Web into feature/testing

noxpardalis committed:
Test if pytest is already included via python environment in Travis

noxpardalis committed:
Check if Travis needs pipenv shell

noxpardalis committed:
Remove manual exit from Travis script

noxpardalis committed:
Add badges for build status

noxpardalis committed:
Rename readme

noxpardalis committed:
Fix table in readme

noxpardalis committed:
Remove testing badge for testing branch

noxpardalis committed:
Fix alignment of table

Commit:
Merge pull request #55 from RUGSoftEng/feature/testing

Feature/testing

andrewlalis committed:
Added marketplace endpoint.

andrewlalis committed:
Added definition of plugin entity.

andrewlalis committed:
Added GET method to list plugin resources.

andrewlalis committed:
Added marketplace to list of namespaces.

andrewlalis committed:
Messed around with the endpoints and stuff.

noxpardalis committed:
Merge remote-tracking branch 'alt/feature/marketplace' into feature/plugin_marketplace

noxpardalis committed:
Remove unnecessary files

noxpardalis committed:
Fix build file

noxpardalis committed:
Move project files to top level

andrewlalis committed:
Merge pull request #1 from RUGSoftEng/feature/plugin_marketplace

Feature/plugin marketplace

andrewlalis committed:
Added start of documentation.

andrewlalis committed:
Updated document, updated gitignore.

andrewlalis committed:
Added more endpoint definitions, example payloads, etc. to doc.

andrewlalis committed:
Added tables schema image.

andrewlalis committed:
Updated gitignore.

andrewlalis committed:
Added plugin class.

andrewlalis committed:
Added tag class, tables file.

andrewlalis committed:
Added uservotes object with constraint on vote variable.

andrewlalis committed:
Added some schemas for accepting input for plugins endpoint.

andrewlalis committed:
Removed unnecessary files, cleaned some things.

andrewlalis committed:
Updated endpoints, working on getting things working with parameters.

andrewlalis committed:
Updated docs to reflect change to url params for get plugins.

andrewlalis committed:
Renamed class files, worked on sql queries.

andrewlalis committed:
Merge pull request #2 from RUGSoftEng/feature/plugin_marketplace

Feature/plugin marketplace

Feikoritsema commented on docs/code/get_plugin_example.json:
```diff
@@ -0,0 +1,12 @@
+	"last_edited_date": "19-05-2018",
+	"version": "0.1.0rc3",
+	"description_short": "A plugin designed to do nothing.",
+	"description_long": "This is a much longer description that is supposed to only be sent when someone really wants to know more about this specific plugin."
```
missing comma

noxpardalis replied:
Fixed.

Feikoritsema commented on docs/code/get_plugins_options.json:
```diff
@@ -0,0 +1,13 @@
+{
+	"count": 10,
+	"author_id": "fhlufkeh74i7y34fh7oh",
+	"title_contains": "example"
```
missing comma

noxpardalis replied:
Fixed.

Feikoritsema commented on api/endpoints/plugins.py:
```diff
@@ -0,0 +1,74 @@
+"""
+Gets a list of plugins as a json string, given some parameters.
+"""
+def get_plugins_with_options(count, page, author_id, title_contains, sort_by, sort_order, tags):
```
title_contains + tags are unused

noxpardalis replied:
Left for extension.

Feikoritsema commented on api/endpoints/plugins.py:
```diff
@@ -0,0 +1,74 @@
+
+
+"""
+Gets a list of plugins as a json string, given some parameters.
```
put this in the method, as PEP8 requires 2 blank lines above the method

noxpardalis replied:
Fixed.

Feikoritsema commented on application.py:
```diff
@@ -0,0 +1,57 @@
+    """
+    Run the backend in production mode.
+    """
+    APP.run(host="0.0.0.0", port=5000)
```
Web already runs on 5000, and use a constant for the port.

noxpardalis replied:
Fixed.

Feikoritsema commented on api/authentication/authentication.py:
```diff
@@ -0,0 +1,178 @@
+
+
+def get_token_auth_header():
+    """Obtains the access token from the Authorization Header
```
formatting

noxpardalis replied:
Fixed.

Feikoritsema commented on api/database/entities/Plugin.py:
```diff
@@ -0,0 +1,57 @@
+    # A longer, more verbose description of the plugin.
+    description_long = Column(Text, nullable=True)
+    # The number of positive votes for a plugin.
+    up_goats = Column(Integer, nullable=False)
```
Why goats and not votes ?

noxpardalis replied:
Paying homage to Amaltheia, the goat who raised Zeus. Ultimately we thought it would be funny for something that won't in its current form leave the prototyping phase.

Feikoritsema commented on tests/test_dummy.py:
```diff
@@ -0,0 +1,3 @@
+import sqlalchemy
```
Pretty useless obviously, if you find the time, try to write some basic tests, shouldn't be too hard.

Feikoritsema reviewed (changes requested):
Looks cool! Very extensive architecture document! Try for the last sprint to improve and add some README for possible contributors in the future.

Style: (PEP8)
- Add newline at the end of file
- Use double quotes for docstrings
- Remove unused imports

noxpardalis committed:
Fix pylint complaints

noxpardalis committed:
Further remedy Feikos comments.

Note that testing has not been tackled as the application would have to be
shifted over to the factory pattern to facilitate DB testing.

noxpardalis committed:
Merge pull request #5 from RUGSoftEng/feature/sprintFix

Feature/sprint fix

noxpardalis committed:
Merge pull request #7 from RUGSoftEng/develop

Develop
